crates/ruff_python_formatter/src/trivia.rs

```diff
@@ -0,0 +1,41 @@
+use ruff_python_ast::whitespace::is_python_whitespace;
+use ruff_text_size::{TextLen, TextRange, TextSize};
+
+/// Searches for the first non-trivia character in `range`.
+///
+/// The search skips over any whitespace and comments.
+///
+/// Returns `Some` if the range contains any non-trivia character. The first item is the absolute offset
+/// of the character, the second item the non-trivia character.
+///
+/// Returns `None` if the range is empty or only contains trivia (whitespace or comments).
+pub(crate) fn find_first_non_trivia_character_in_range(
```

Comment:
I extracted this utility because we can re-use it when e.g. searching for a trailing `,`. 

We probably want another utility that searches backwards. This is going to be more expensive if the range contains line breaks, because we need to run to the start of the line to verify that the non-trivia character isn't a comment. However, we only need to do this for the second line:

```python
(a) # Cheap, because no line break between `(` and `a`
(
	# this is a comment
	a
) # Needs to process all characters, including the comment. 

(b + 
	a
) # Must process all of `(b +` only to return `+` the plus when searching from the `a` because it must test if `(b + ` is a comment
```